app/main.py

```diff
@@ -2,0 +1,36 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{"
+                f"Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}"
+                f"}}")
+
+    @classmethod
+    def __str__(cls) -> str:
+        return "[" + ", ".join(repr(animal) for animal in cls.alive) + "]"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Animal) -> None:
```

Comment:
The `bite` method should ideally accept a parameter of type `Herbivore` instead of `Animal` to ensure type safety, as the method logic specifically checks for `Herbivore` instances.